src/v/config/configuration.cc

```diff
@@ -2128,6 +2128,17 @@ configuration::configuration()
       "cloud_storage_azure_adls_endpoint.",
       {.needs_restart = needs_restart::yes, .visibility = visibility::user},
       std::nullopt)
+  , cloud_storage_azure_hierarchical_namespace_enabled(
+      *this,
+      "cloud_storage_azure_hierarchical_namespace_enabled",
+      "Whether or not Azure Hierarchical Namespaces are enabled on the "
+      "cloud_storage_azure_storage_account. If this property is not set, "
+      "cloud_storage_azure_shared_key must be set, and each node will try to "
```

Comment:
> If this property is not set, cloud_storage_azure_shared_key must be set

Is `cloud_storage_azure_shared_key` mandatory? From reading the test code it doesn't seem so. It is just an override and can be left nullopt for auto inference in both cases.

Reply:
There is not test covering this property. ⚠️

Reply:
> is cloud_storage_azure_shared_key mandatory

it was mandatory before the introduction of cloud_storage_credential_sources = azure_aks_oidc_federation / azure_vm_instance_metadata

> There is not test covering this property. ⚠️

test_abs_cloud_validation (in this PR) is a positive test for valid configurations, but I'll add a negative test